planet/cli/data.py

```diff
@@ -0,0 +1,118 @@
+"""The Planet Data CLI."""
+
+import json
+from typing import AsyncIterator, List
+
+import click
+
+from ..exceptions import PlanetError
+from .cmds import coro
+
+# Parameter callbacks are what we use to validate and transform the
+# values of arguments and options. On the command line, everything is
+# a string, but we want to transform to natural Python types as soon as
+# possible.
```

Comment:
Boundaries are important. Our CLI interoperates with other programs using text. One kind of text is a comma-separated string of asset item types. Another kind of text is a JSON-encoded set of filters. But inside our CLI, a different language is spoken and we use built in or custom Python types. A `list` of asset item types with no comma separators. A filter object (a `dict`) decoded from JSON.